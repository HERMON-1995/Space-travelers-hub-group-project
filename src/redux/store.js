import thunk from 'redux-thunk';
import { legacy_createStore as createStore, combineReducers, applyMiddleware } from '@reduxjs/toolkit';
import rockets from './rocketsSlice';
import mission from './missionsSlice';

const reducers = combineReducers({
  rocketsReducer: rockets,
  missionReducer: mission,
});

const store = createStore(reducers, applyMiddleware(thunk));

export default store;
