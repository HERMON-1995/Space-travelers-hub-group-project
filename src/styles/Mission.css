.mission-container {
  width: 100%;
}

table {
  width: 90%;
  margin: 1% auto;
  border-collapse: collapse;
  padding: 0;
}

th {
  margin: 0;
  border: 1px solid #cac9c9;
}

td {
  text-align: left;
  border: 1px solid #cac9c9;
  padding: 1%;
}

.description {
  width: 70%;
}

table tr:nth-child(even) {
  background-color: rgb(212, 212, 212);
}

.header {
  text-align: left;
  padding-left: 1%;
}
