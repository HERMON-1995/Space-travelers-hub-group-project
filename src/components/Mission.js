import React from 'react';
import '../styles/Mission.css';
import { useSelector, useDispatch } from 'react-redux';
import { reservemission, cancelmission } from '../redux/missionsSlice';

const Mission = () => {
  const { missionList } = useSelector((state) => state.missionReducer);
  const dispatch = useDispatch();

  const memberBtn = {
    color: 'White',
    backgroundColor: 'red',
    fontWeight: '300',
    padding: '3%',
    borderRadius: '10%',
    border: 'none',
  };
  const ntmemberBtn = {
    color: 'White',
    backgroundColor: '#787474',
    fontWeight: '300',
    padding: '3%',
    borderRadius: '10%',
    border: 'none',
  };

  const leaveBtn = {
    color: 'red',
    backgroundColor: 'transparent',
    padding: '5%',
    border: 'solid 1px red',
  };

  const joinBtn = {
    color: '#787474',
    border: 'solid 1px #787474',
    backgroundColor: 'transparent',
    padding: '5%',
  };
  const tbstyle = { border: '1px solid #cac9c9' };
  const tbDesction = {
    width: '55%',
    textAlign: 'justify',
    border: '1px solid #cac9c9',
  };
  return (
    <div className="mission-container">
      <table>
        <tbody>
          <tr>
            <th className="header" style={tbstyle}>Mission</th>
            <th className="header" style={tbstyle}>Description</th>
            <th className="header" style={tbstyle}>Status</th>
            <th className="header" style={tbstyle} aria-label="jheader" />
          </tr>
          {missionList.map((data) => {
            const value = data.missionValue;
            return (
              <tr key={data.id}>
                <td style={tbstyle}>{data.missionName}</td>
                <td style={tbDesction} className="description">{data.missionDescription}</td>
                <td style={tbstyle}>
                  <button style={value ? memberBtn : ntmemberBtn} aria-label="member" type="submit">{data.memberShip}</button>
                </td>
                <td style={tbstyle}>
                  { !data.missionValue
                    ? (
                      <button style={value ? leaveBtn : joinBtn} className="btndecision" aria-label="join" type="submit" onClick={() => dispatch(reservemission(data.id))}>
                        {' '}
                        {data.missionStatus}
                        {' '}
                      </button>
                    ) : (
                      <button style={value ? leaveBtn : joinBtn} className="btndecision" aria-label="join" type="submit" onClick={() => dispatch(cancelmission(data.id))}>
                        {' '}
                        {data.missionStatus}
                        {' '}
                      </button>
                    )}
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};

export default Mission;
