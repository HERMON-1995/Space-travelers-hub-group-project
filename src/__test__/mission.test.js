import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Mission from '../components/Mission';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Mission />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});