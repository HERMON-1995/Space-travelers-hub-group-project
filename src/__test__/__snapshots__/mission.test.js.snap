// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mission-container"
>
  <table>
    <tbody>
      <tr>
        <th
          className="header"
          style={
            Object {
              "border": "1px solid #cac9c9",
            }
          }
        >
          Mission
        </th>
        <th
          className="header"
          style={
            Object {
              "border": "1px solid #cac9c9",
            }
          }
        >
          Description
        </th>
        <th
          className="header"
          style={
            Object {
              "border": "1px solid #cac9c9",
            }
          }
        >
          Status
        </th>
        <th
          aria-label="jheader"
          className="header"
          style={
            Object {
              "border": "1px solid #cac9c9",
            }
          }
        />
      </tr>
    </tbody>
  </table>
</div>
`;
